### PR TITLE
KAFKA-18026: KIP-1112, configure all StoreBuilder & StoreFactory layers

### PR DESCRIPTION
**Pull request by ableegoldman**
Because of how we have to wrap StoreFactory and StoreBuilder layers on top of each other for various parts of the topology building process, we need to make sure both of these are capable of configuration and will check for & delegate to an underlying layer if it exists

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreBuilderWrapper.java [hunk 1]
```diff
@@ -34,6 +35,7 @@
  */
 public class StoreBuilderWrapper implements StoreFactory {
 
+
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreBuilderWrapper.java:38]: nit? typo.

**Review by guozhangwang — APPROVED** [review]
LGTM.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreBuilderWrapper.java (hunk 1 above)
Comment (ableegoldman) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreBuilderWrapper.java:38]: yeah, typo. I'll remove this in one of my followup PRs so I can go ahead and merge this guy now

**Comment by ableegoldman** [comment]
Merged to trunk and cherrypicked to 4.0

**Pull request merged**